### PR TITLE
New revstat filter strings

### PR DESCRIPTION
**Pull request by dnil**
This PR adds a functionality or fixes a bug.
OR
This PR marks a new Scout release. We apply semantic versioning. This is a major/minor/patch release for reasons.

**How to test**:
1. how to test it, possibly with real cases/data

**Expected outcome**:
The functionality should be working
Take a screenshot and attach or copy/paste the output.

**Review:**
- [x] code approved by CR
- [x] tests executed by CR, DN

### CODE REVIEW

**Comment by northwestwitch** [comment]
I'm going to write a specific test that uses the clinical filter to get variants which have new revstat criteria.

**Comment by northwestwitch** [comment]
Possible testing for this PR: 
- Take this demo variant http://localhost:5000/cust000/643594/d57f0b1933560335fca26e6dd687e60c and add the following clnsig criteria:
```
clinsig_criteria = {
        "value" : 5,
        "accession" : 345986,
        "revstat" : "criteria_provided,multiple_submitters,no_conflicts"
    }
```
Variant should show up when clinical filter is called (because the other variant specifics for clinical filter are also met, I think)

**Comment by northwestwitch** [comment]
I'm trying the clinical filter query with the tested suggested in the comment above, so far I didn't have much success. The clinical filter query looks quite complicated and I wonder if it can be simplified..

Now it is like this:
```
{
  'case_id': 'internal_id', 
  'category': 'snv', 
  'variant_type': 'clinical', 
  'panels': {'$in': ['panel1']}, 
  '$or': [
    {'$and': [
      {'$or': [
        {'gnomad_frequency': {'$lt': 0.01}}, 
        {'gnomad_frequency': {'$exists': False}}]}, 
        {'genes.functional_annotation': {'$in': ['transcript_ablation', 'splice_donor_variant', 'splice_acceptor_variant', 'stop_gained', 'frameshift_variant', 'stop_lost', 'start_lost', 'initiator_codon_variant', 'inframe_insertion', 'inframe_deletion', 'missense_variant', 'protein_altering_variant', 'transcript_amplification', 'splice_region_variant', 'incomplete_terminal_codon_variant', 'synonymous_variant', 'start_retained_variant', 'stop_retained_variant', 'coding_sequence_variant']}},
        {'genes.region_annotation': {'$in': ['exonic', 'splicing']}}]}, 
        {'clnsig': 
          {'$elemMatch': 
            {'$or': [
              {'$and': [
                {'value': {'$in': [4, 'Likely pathogenic', 5, 'Pathogenic']}}, 
                {'revstat': {'$in': ['mult', 'multiple_submitters', 'single', 'single_submitter', 'exp', 'reviewed_by_expert_panel', 'guideline', 'practice_guideline']}}]}, 
                {'$and': [
                  {'value': re.compile('Likely pathogenic|Pathogenic')}, 
                  {'revstat': re.compile('mult|multiple_submitters|single|single_submitter|exp|reviewed_by_expert_panel|guideline|practice_guideline')}
                  ]
                }
              ]
            }
          }
        }
    ]
}
```

**Comment by dnil** [comment]
Sorry, just back from meeting. Right, it is a bit complex, and will probably remain so, but sure there could be simplifications! Lets look. It is supposed to be of the this - but also this type. There are two distinct populations that we want to let through: 1) those that fulfil a certain set of frequency and predicted severity criteria and 2) those that have previous classifications of clinical evidence, regardless of 1. And both while retaining the mask of the active (clinical) gene panel selection.

**Comment by dnil** [comment]
As it is now it for instance has this extra `$and': [`from the query generation. I suppose that is from some features from the more general queries that we don't use in the clinical filter, but lets look!

**Comment by dnil** [comment]
Oh, and it is doing those re:s and $in matching? Why is that?

**Comment by dnil** [comment]
Right, the int valued ones.

**Comment by northwestwitch** [comment]
I'm proposing a fix to the query.
I thing the clnsig part of the query should be like this:
```
    clnsig_query = {
                "clnsig": {
                    "$elemMatch": {
                        "$or": [
                            {
                                "$and": [
                                    {"value": {"$in": rank}},
                                    {'$or': [
                                        {"revstat": {"$in": trusted_revision_level}},
                                        {"revstat": re.compile("|".join(trusted_revision_level))}
                                    ]}
                                ]
                            },
                            {
                                "$and": [
                                    {"value": re.compile("|".join(str_rank))},
                                    {'$or': [
                                        {"revstat": {"$in": trusted_revision_level}},
                                        {"revstat": re.compile("|".join(trusted_revision_level))}
                                    ]}
                                ]
                            },
                        ]
                    }
                }
            }
```

**Comment by northwestwitch** [comment]
Testing after my fix to the clinsig part of the query, The modified gene shows up!

Master branch:

image


This branch:
image

**Comment by dnil** [comment]
Agreed; no use in having both $in and re for the same string. Since this is a re match, a few wildcards would shorten it a bit as well; e.g. mult*, but that would also require more careful checking of what strings are coming in from the CSQ..

**Comment by dnil** [comment]
Also, the revstat part is the same for both string and rank types now and could be broken out!

**Comment by northwestwitch** [comment]
> no use in having both $in and re for the same string.

So you would remove the $in?

**Comment by dnil** [comment]
no, I'd keep the $in for the sake of the ints, unless you are very sure of the internal str - int conversion in python to mongo..

**Comment by northwestwitch** [comment]
> unless you are very sure of the internal str - int conversion in python to mongo..

I'm not sure about anything when it comes to queries! 😆

**Comment by northwestwitch** [comment]
Ok, I'll try to fix the tests

**Comment by dnil** [comment]
I can try to break out the revstat-section so we only use it once!

**Comment by dnil** [comment]
Like so: 
`clnsig_query = {
                "clnsig": {
                    "$elemMatch": {
                        "$and": [
                            {
                                "$or": [
                                    {"value": {"$in": rank}},
                                    {"value": re.compile("|".join(str_rank))},
                                ]
                            },
                            {"$or": [
                                {"revstat": {"$in": trusted_revision_level}},
                                {"revstat": re.compile("|".join(trusted_revision_level))}
                            ]}
                        ]
                    }
                }
            }
`

**Comment by dnil** [comment]
Even better: 
```
            clnsig_query = {
                "clnsig": {
                    "$elemMatch": {
                        "$and": [
                            {"$or": [
                                    {"value": {"$in": rank}},
                                    {"value": re.compile("|".join(str_rank))},
                                ]
                            },
                            {"$or": [
                                {"revstat": re.compile("|".join(trusted_revision_level))}
                            ]}
                        ]
                    }
                }
            }
```

**Comment by dnil** [comment]
The revstats don't need an in when they are being re:d; no int levels there.

**Comment by northwestwitch** [comment]
Looks like a better solution, please push it to the branch!

**Comment by dnil** [comment]
You were right! 😉

**Comment by northwestwitch** [comment]
Testing your latest changes, it works:

image

**Comment by northwestwitch** [comment]
> You were right! 😉

About what?

**Comment by dnil** [comment]
> About what?

It could be simplified! 👍

**Comment by dnil** [comment]
Ok, lets change the test then. Do you have a version already, or should I?

**Comment by northwestwitch** [comment]
> Ok, lets change the test then. Do you have a version already, or should I?

You could fix the test_query ones and I can fix the test_variants_view with the clinical filter?

File: tests/server/blueprints/variants/test_variants_views.py [hunk 1]
```diff
@@ -1,31 +1,66 @@
...
+            ),
+            data=data,
+            content_type="application/x-www-form-urlencoded",
         )
+
         # THEN it should return a page
         assert resp.status_code == 200
 
+        # containing the variant above
+        assert updated_var["_id"] in str(resp.data)
```
Comment (dnil) [inline tests/server/blueprints/variants/test_variants_views.py:60]: :+1:

File: tests/server/blueprints/variants/test_variants_views.py [hunk 2]
```diff
@@ -58,22 +90,19 @@ def test_variants_post_data(app, institute_obj, case_obj):
         resp = client.get(url_for("auto_login"))
         assert resp.status_code == 200
 
-        # WHEN submitting form data to the page (POST method)
-        data = urlencode({"clinical_filter": True})  # clinical filter
-
-        # WHEN requesting a clinical filter to variants page
-        resp = client.post(
+        # WHEN accessing the variants page
+        resp = client.get(
```
Comment (dnil) [inline tests/server/blueprints/variants/test_variants_views.py:94]: 👍

**Pull request merged**